src/argilla/client/users.py

```diff
@@ -379,8 +379,6 @@ def list(cls) -> Iterator["User"]:
         """
         client = cls.__active_client()
         try:
-            users = users_api.list_users(client).parsed
-            for user in users:
-                yield cls.__new_instance(client, user)
+            return users_api.list_users(client).parsed
```

Comment:
Note that you're not returning a `User` here, but the `User` model from the SDK, which won't come with the methods and properties of the class, so this should be rewritten as

```suggestion
            return [cls.__new_instance(client, user) for user in users_api.list_users(client).parsed]
```

Reply:
Thanks for noticing!